### PR TITLE
Editor and View should be ReferencedComposite

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

**Comment by rawagner** [comment]
testPR

**Comment by rawagner** [comment]
testPR

**Comment by rawagner** [comment]
testPR

**Comment by apodhrad** [comment]
**Error Message**
```
No matching widget found with Matcher matching when all matchers match: [Matcher matching widget with the same type as or type extending class org.eclipse.swt.custom.CTabItem, Matcher matching widgets with text that without mnenomic matches: is "Source"]
class org.eclipse.e4.ui.workbench.renderers.swt.ContributedPartRenderer$2[shell: 'data - Java - xml-project/file.xml - Eclipse Platform']
 class org.eclipse.swt.widgets.Composite[shell: 'data - Java - xml-project/file.xml - Eclipse Platform']
  class org.eclipse.swt.widgets.Tree[shell: 'data - Java - xml-project/file.xml - Eclipse Platform']
```
**Stacktrace**
```
org.eclipse.reddeer.core.exception.CoreLayerException: No matching widget found with Matcher matching when all matchers match: [Matcher matching widget with the same type as or type extending class org.eclipse.swt.custom.CTabItem, Matcher matching widgets with text that without mnenomic matches: is "Source"]
class org.eclipse.e4.ui.workbench.renderers.swt.ContributedPartRenderer$2[shell: 'data - Java - xml-project/file.xml - Eclipse Platform']
	class org.eclipse.swt.widgets.Composite[shell: 'data - Java - xml-project/file.xml - Eclipse Platform']
		class org.eclipse.swt.widgets.Tree[shell: 'data - Java - xml-project/file.xml - Eclipse Platform']

	at org.eclipse.reddeer.common.wait.WaitUntil.throwWaitTimeOutException(WaitUntil.java:108)
	at org.eclipse.reddeer.common.wait.AbstractWait.timeoutExceeded(AbstractWait.java:190)
	at org.eclipse.reddeer.common.wait.AbstractWait.wait(AbstractWait.java:135)
	at org.eclipse.reddeer.common.wait.AbstractWait.<init>(AbstractWait.java:116)
	at org.eclipse.reddeer.common.wait.AbstractWait.<init>(AbstractWait.java:81)
	at org.eclipse.reddeer.common.wait.AbstractWait.<init>(AbstractWait.java:63)
	at org.eclipse.reddeer.common.wait.WaitUntil.<init>(WaitUntil.java:49)
	at org.eclipse.reddeer.core.lookup.WidgetLookup.activeWidget(WidgetLookup.java:105)
	at org.eclipse.reddeer.core.lookup.WidgetLookup.activeWidget(WidgetLookup.java:84)
	at org.eclipse.reddeer.swt.widgets.AbstractWidget.<init>(AbstractWidget.java:72)
	at org.eclipse.reddeer.swt.widgets.AbstractItem.<init>(AbstractItem.java:30)
	at org.eclipse.reddeer.swt.impl.ctab.AbstractCTabItem.<init>(AbstractCTabItem.java:38)
	at org.eclipse.reddeer.swt.impl.ctab.DefaultCTabItem.<init>(DefaultCTabItem.java:97)
	at org.eclipse.reddeer.swt.impl.ctab.DefaultCTabItem.<init>(DefaultCTabItem.java:57)
	at org.eclipse.reddeer.swt.impl.ctab.DefaultCTabItem.<init>(DefaultCTabItem.java:47)
	at org.eclipse.reddeer.eclipse.ui.part.MultiPageEditorPart.selectPage(MultiPageEditorPart.java:88)
	at org.eclipse.reddeer.eclipse.test.ui.part.MultiPageEditorTest.selectPage(MultiPageEditorTest.java:75)
```

File: plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/api/WorkbenchPart.java [hunk 1]
```diff
@@ -10,34 +10,22 @@
...
 	
 	/**
 	 * Activates workbench part.
 	 */
 	void activate();
 
 	/**
-	 * Minimize workbench part.
-	 */
-	void minimize();
```
Comment (apodhrad) [inline plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/api/WorkbenchPart.java]: Why did you remove this method?

File: plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/condition/ActiveFocusControlIsInActiveView.java [hunk 2]
```diff
@@ -26,13 +26,13 @@
 public class ActiveFocusControlIsInActiveView extends AbstractWaitCondition {
 	
 	/* (non-Javadoc)
-	 * @see org.eclipse.reddeer.common.condition.WaitCondition#test()
+	 * @see org.jboss.reddeer.common.condition.WaitCondition#test()
```
Comment (apodhrad) [inline plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/condition/ActiveFocusControlIsInActiveView.java]: There should org.eclipse instead of org.jboss.

File: plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/condition/ActiveFocusControlIsInActiveView.java [hunk 3]
```diff
@@ -50,7 +50,7 @@ public Boolean run() {
 	}
 	
 	/* (non-Javadoc)
-	 * @see org.eclipse.reddeer.common.condition.AbstractWaitCondition#description()
+	 * @see org.jboss.reddeer.common.condition.AbstractWaitCondition#description()
```
Comment (apodhrad) [inline plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/condition/ActiveFocusControlIsInActiveView.java]: Again org.eclipse vs org.jboss

File: plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/handler/EditorHandler.java [hunk 4]
```diff
@@ -76,41 +75,6 @@ public void run() {
     }
 
     /**
-     * Checks if editor is active.
-     * @param editor to be checked if it is active
-     * @return true if editor is active, false otherwise
-     */
-    public boolean isActive(final IEditorPart editor) {
```
Comment (apodhrad) [inline plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/handler/EditorHandler.java]: Why did you remove this method?

File: plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/handler/EditorHandler.java [hunk 5]
```diff
@@ -76,41 +75,6 @@ public void run() {
...
-                        .getActivePart());
-            }
-        });
-    }
-
-    /**
-     * Activates editor.
-     * @param editor to activate
-     */
-    public void activate(final IEditorPart editor) {
```
Comment (apodhrad) [inline plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/handler/EditorHandler.java]: Why did you remove this method?

Comment (rawagner) [inline plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/handler/EditorHandler.java]: it is already defined in WorkbenchPartHandler

File: plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/handler/EditorHandler.java (hunk 4 above)
Comment (rawagner) [inline plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/handler/EditorHandler.java]: it is already defined in WorkbenchPartHandler

**Comment by rawagner** [comment]
@apodhrad minimize/maximize/restore is back. I also removed jboss references

File: plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/impl/table/AbstractTable.java [hunk 6]
```diff
@@ -196,7 +196,7 @@ public void deselectAll() {
 	}
 
 	private void waitUntilTableHasRows() {
-		new WaitUntil(new TableHasRows(this), TimePeriod.DEFAULT, false);
+		new WaitUntil(new TableHasRows(this), TimePeriod.SHORT, false);
```
Comment (apodhrad) [inline plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/impl/table/AbstractTable.java]: Is it enough?

Comment (rawagner) [inline plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/impl/table/AbstractTable.java]: IMHO yes, if somebody has a table which takes longer to init data, then he should use WaitUntil in his code. 
If I have a table which has 0 items and I will call table.getItems() I will always have to wait for DEFAULT time period (3s) which is quite long.
